Slack.Automation/Promact.Core.Repository/TaskMailRepository/TaskMailRepository.cs

```diff
@@ -359,804 +382,439 @@ public TaskMailRepository(IRepository<TaskMail> taskMail, IStringConstantReposit
-                        UserId = UserId,
-                        UserName = UserName,
-                        UserRole = UserRole,
-                        CreatedOn = Convert.ToDateTime(CreatedOn).Date,
-                        TaskMails = taskMailReport,
-                        IsMin = Convert.ToDateTime(minDate).Date,
-                        IsMax = Convert.ToDateTime(maxDate).Date
-
-                    };
-                    taskMailAc.Add(taskMailUserAc);
+                    List<TaskMailDetailReportAc> taskMailDetailReportList = new List<TaskMailDetailReportAc>();
+                    var TaskMailDetailReportAc = new TaskMailDetailReportAc(description: _stringConstant.NotAvailable, comment: _stringConstant.NotAvailable, status: TaskMailStatus.none);
+                    taskMailDetailReportList.Add(TaskMailDetailReportAc);
+                    TaskMailReportAc taskMailReportAc = new TaskMailReportAc(userId, role, userName, taskMailDetailReportList, createdOn:Convert.ToDateTime(createdOn).Date, maxDate:  Convert.ToDateTime(maxDate).Date, minDate: Convert.ToDateTime(minDate).Date);
+                    taskMailReportAcList.Add(taskMailReportAc);
                 }
             }
             else
             {
-                List<TaskMailReportAc> taskMailReport = new List<TaskMailReportAc>();
-                TaskMailReportAc taskmailReportAcTL = new TaskMailReportAc
-                {
-                    Id = 0,
-                    Description = _stringConstant.NotAvailable,
-                    Comment = _stringConstant.NotAvailable,
-                    Hours = 0
-                };
-                taskMailReport.Add(taskmailReportAcTL);
-                TaskMailUserAc taskMailUserAc = new TaskMailUserAc
-                {
-                    UserId = UserId,
-                    UserName = UserName,
-                    UserRole = UserRole,
-                    CreatedOn = Convert.ToDateTime(CreatedOn).Date,
-                    TaskMails = taskMailReport,
-                    IsMin = Convert.ToDateTime(CreatedOn).Date,
-                    IsMax = Convert.ToDateTime(CreatedOn).Date
-
-                };
-                taskMailAc.Add(taskMailUserAc);
+                List<TaskMailDetailReportAc> taskMailDetailReportList = new List<TaskMailDetailReportAc>();
+                var TaskMailDetailReportAc = new TaskMailDetailReportAc(description: _stringConstant.NotAvailable, comment: _stringConstant.NotAvailable, status: TaskMailStatus.none);
+                taskMailDetailReportAcList.Add(TaskMailDetailReportAc);
+                TaskMailReportAc taskMailReportAc = new TaskMailReportAc(userId, role, userName, taskMailDetailReportList, createdOn: Convert.ToDateTime(createdOn).Date, maxDate:Convert.ToDateTime(createdOn).Date,minDate: Convert.ToDateTime(createdOn).Date);
+                taskMailReportAcList.Add(taskMailReportAc);
             }
-            return taskMailAc;
+            return taskMailReportAcList;
```

Comment:
Almost all methods in TaskMailRepository is too long and redundant. Please optimize code